installer/datafiles/Linux.data

```diff
@@ -16,11 +16,21 @@ ULINUX
 
 %Preinstall_50
 # VerifySSLVersion
-openssl version | awk '{print $2}' | grep -Eq '^0.9.8|^1.0.'
-if [ $? -ne 0 ]; then
-    echo 'Unsupported OpenSSL version - must be either 0.9.8* or 1.0.*.'
-    echo 'Installation cannot proceed.'
-    exit 1
+OPENSSL_SYSTEM_VERSION_FULL=`openssl version | awk '{print $2}'`
+OPENSSL_SYSTEM_VERSION_098=`echo $OPENSSL_SYSTEM_VERSION_FULL | grep -Eq '^0.9.8'; echo $?`
+OPENSSL_SYSTEM_VERSION_100=`echo $OPENSSL_SYSTEM_VERSION_FULL | grep -Eq '^1.0.'; echo $?`
+[ `uname -m` = "x86_64" ] && OPENSSL_SYSTEM_VERSION_110=`echo $OPENSSL_SYSTEM_VERSION_FULL | grep -Eq '^1.1.'; echo $?`
+
+if [ $OPENSSL_SYSTEM_VERSION_098 = 1 -a $OPENSSL_SYSTEM_VERSION_100 = 1 ];then 
+    if [ $OPENSSL_SYSTEM_VERSION_110 = 1 ];then
+        echo 'Unsupported OpenSSL version - must be either 0.9.8* or 1.0.*. or 1.1.*.'
+        echo 'Installation cannot proceed.'
+        exit 1
+    elif [ $OPENSSL_SYSTEM_VERSION_110 != 0 ];then
+        echo 'Unsupported OpenSSL version - must be either 0.9.8* or 1.0.*.'
+        echo 'Installation cannot proceed.'
+        exit 1
+    fi
```

Comment:
Hmm. I guess that works, but it feels longer and more involved. You have the output of `uname -m` on the destination system. Why not just throw (old) lines 19-23 into an `if` clause: If x86_64, check all three (at once), else do lines 19-23. Seems more straightforward to me, but I'm thinking you had a reason for that which I didn't consider?

Reply:
This approach also looks fine to me but I think code replication will be more. I will revisit this code.

Reply:
Fixed